idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/proofs/intentions/coercion.kt

```diff
@@ -0,0 +1,80 @@
+package arrow.meta.ide.plugins.proofs.intentions
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.log.Log
+import arrow.meta.log.invoke
+import arrow.meta.phases.CompilerContext
+import arrow.meta.phases.ExtensionPhase
+import arrow.meta.phases.resolve.baseLineTypeChecker
+import arrow.meta.plugins.proofs.phases.coerceProof
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.debugger.sequence.psi.resolveType
+import org.jetbrains.kotlin.jsr223.KotlinJsr223StandardScriptEngineFactory4Idea
+import org.jetbrains.kotlin.nj2k.postProcessing.type
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtElement
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.descriptorUtil.module
+import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
+
+fun IdeMetaPlugin.coercionIntention(): ExtensionPhase =
+  addIntention(
+    text = "Make coercion explicit",
+    kClass = KtElement::class.java,
+    isApplicableTo = { ktCall: KtElement, caretOffset: Int ->
+      when (ktCall) {
+        is KtDotQualifiedExpression -> ktCall.isDotQualifiedExpressionCoerced()
+        is KtProperty -> ktCall.isCoerced()
+        else -> false
+      }
+    },
+    applyTo = { ktCall, editor ->
+      // replace with proof previously found
```

Comment:
This is where we'd provide the quickfix?

Reply:
Exactly, although the name of the function is not straightforward to grasp IMO

Reply:
Yeah! in `isApplicableTo` we look for the KtElements that we wanna offer the options to change and in `applyTo` the fixes that can be applied 👍